geti_sdk/deployment/deployed_model.py

```diff
@@ -515,13 +515,19 @@ def infer(self, image: np.ndarray, explain: bool = False) -> Prediction:
 
         # Add optional explainability outputs
         if explain:
-            if hasattr(postprocessing_results, "feature_vector"):
-                prediction.feature_vector = postprocessing_results.feature_vector
-            result_medium = ResultMedium(name="saliency map", type="saliency map")
-            result_medium.data = self._converter.convert_saliency_map(
-                postprocessing_results, image_shape=image.shape
-            )
-            prediction.maps.append(result_medium)
+            if self.has_xai_head:
+                if hasattr(postprocessing_results, "feature_vector"):
+                    prediction.feature_vector = postprocessing_results.feature_vector
+                result_medium = ResultMedium(name="saliency map", type="saliency map")
+                result_medium.data = self._converter.convert_saliency_map(
+                    postprocessing_results, image_shape=image.shape
+                )
+                prediction.maps.append(result_medium)
+            else:
+                logging.warning(
```

Comment:
I'm a bit on the fence about this one, but I think I'd prefer to raise an error in this case. I know the prediction will still be available, but if a user calls `Deployment.explain()` most likely they will require the saliency maps down the line. If we only log a warning, downstream code may break and it will be harder to debug

Reply:
changed to exception